src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -209,3 +231,35 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
+register_cli_argument('snapshot', 'snapshot_name', existing_snapshot_name, id_part='name', completer=get_resource_name_completion_list('Microsoft.Compute/snapshots'))
+register_cli_argument('snapshot', 'name', arg_type=name_arg_type)
+register_cli_argument('snapshot', 'storage_account_type', arg_type=disk_underlined_storage_account_type)
+
+existing_image_name = CliArgumentType(overrides=name_arg_type, help='The name of the custom image', completer=get_resource_name_completion_list('Microsoft.Compute/images'), id_part='name')
+register_cli_argument('image', 'os_type', **enum_choice_list(['Windows', 'Linux']))
+register_cli_argument('image', 'image_name', arg_type=name_arg_type, id_part='name', completer=get_resource_name_completion_list('Microsoft.Compute/images'))
+register_cli_argument('image create', 'name', arg_type=name_arg_type, validator=process_image_create_namespace, help='new image name')
+register_cli_argument('image create', 'os_blob_uri', help='blob uri of os virtual hard disk')
+register_cli_argument('image create', 'data_blob_uris', arg_type=multi_ids_type, help='space separated blob uris of data virtual hard disk')
+register_cli_argument('image create', 'os_snapshot', help='snapshot id or name to copy for the os disk')
+register_cli_argument('image create', 'os_disk', help='managed os disk id or name to copy for the os disk')
+register_cli_argument('image create', 'data_disks', arg_type=multi_ids_type, help='space separated managed disk id or name to copy for datat disks')
+register_cli_argument('image create', 'data_snapshots', arg_type=multi_ids_type, help='space separated snapshot id or name to copy for datat disks')
+register_cli_argument('image create', 'source_virtual_machine', help='id or name of a generalized virtual machine to capture image from',
+                      completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachines'))
+
+for scope in ['disk', 'snapshot']:
+    register_cli_argument(scope, 'source_blob_uri', help='a sas uri to a blob to be imported into a managed disk or snapshot')
+    register_cli_argument(scope, 'source_disk', help='id or name of the source managed disk to copy from', validator=process_disk_or_snapshot_create_namespace,
+                          completer=get_resource_name_completion_list('Microsoft.Compute/disks'))
+    register_cli_argument(scope, 'source_snapshot', help='id or name of the source snapshot to copy from', validator=process_disk_or_snapshot_create_namespace,
+                          completer=get_resource_name_completion_list('Microsoft.Compute/snapshots'))
+    register_cli_argument(scope, 'size_gb', options_list=('--size-gb', '-z'), help='size in GB.')
+    register_cli_argument(scope, 'duration_in_seconds', help='Time duration in seconds until the SAS access expires')
```

Comment:
`duration_in_seconds` is reviewed by CRP and we will keep. `--size-gb` is being discussed.